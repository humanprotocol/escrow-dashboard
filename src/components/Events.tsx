import * as React from 'react';
import { Divider, Typography } from '@mui/material';
import Link from '@mui/material/Link';
import Box from '@mui/material/Box';
import { INDENT } from 'src/ui';

interface IEvents {
  url: string;
  scanner: string;
}

const Events: React.FC<IEvents> = ({ url, scanner }): React.ReactElement => {
  return (
    <Box textAlign="center">
      <Divider sx={{ mt: INDENT.S }}>
        <Typography variant="body2" color="text.secondary">
          All deployed escrows
        </Typography>
      </Divider>
      <Link href={url} target="_blank" rel="noreferrer" align="center">
        {scanner}
      </Link>
      <Typography
        variant="body2"
        color="text.secondary"
        sx={{
          fontSize: 11,
        }}
      >
        Each event has a payload of ERC20 token address and Escrow Address
      </Typography>
      <Typography
        variant="body2"
        color="text.secondary"
        sx={{
          fontSize: 11,
        }}
      >
        Change the type of the second argument to "Address" to see an Escrow
        address
      </Typography>
    </Box>
  );
};

export default Events;
