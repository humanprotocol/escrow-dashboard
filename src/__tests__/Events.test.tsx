import * as React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Events from '../components/Events';

const mock = {
  url: 'Url',
  scanner: 'TestScanner',
  text1: 'All deployed escrows',
  text2: 'Each event has a payload of ERC20 token address and Escrow Address',
  text3:
    'Change the type of the second argument to "Address" to see an Escrow address',
};

describe('when rendered Events component', () => {
  it('should render passed prop url', () => {
    render(<Events url={mock.url} scanner={mock.scanner} />);
    expect(screen.findByLabelText(mock.url)).toBeTruthy();
  });

  it('should render passed prop scanner', () => {
    render(<Events url={mock.url} scanner={mock.scanner} />);
    expect(screen.findByLabelText(mock.scanner)).toBeTruthy();
  });

  it('should render prop text1', () => {
    render(<Events url={mock.url} scanner={mock.scanner} />);
    expect(screen.findByLabelText(mock.text1)).toBeTruthy();
  });

  it('should render prop text2', () => {
    render(<Events url={mock.url} scanner={mock.scanner} />);
    expect(screen.findByLabelText(mock.text2)).toBeTruthy();
  });

  it('should render prop text3', () => {
    render(<Events url={mock.url} scanner={mock.scanner} />);
    expect(screen.findByLabelText(mock.text3)).toBeTruthy();
  });
});

it('renders correctly', () => {
  const tree = renderer
    .create(<Events url={mock.url} scanner={mock.scanner} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
