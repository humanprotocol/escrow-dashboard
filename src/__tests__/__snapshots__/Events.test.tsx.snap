// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events component renders correctly, corresponds to the snapshot 1`] = `
<div
  className="MuiBox-root css-xi606m"
>
  <div
    className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-n4pdsy-MuiDivider-root"
    role="separator"
  >
    <span
      className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
      >
        All deployed escrows
      </p>
    </span>
  </div>
  <a
    className="MuiTypography-root MuiTypography-inherit MuiTypography-alignCenter MuiLink-root MuiLink-underlineAlways css-8wztv5-MuiTypography-root-MuiLink-root"
    href="Url"
    onBlur={[Function]}
    onFocus={[Function]}
    rel="noopener noreferrer"
    target="_blank"
  >
    TestScanner
  </a>
  <p
    className="MuiTypography-root MuiTypography-body2 css-1ncpc4g-MuiTypography-root"
  >
    Each event has a payload of ERC20 token address and Escrow Address
  </p>
  <p
    className="MuiTypography-root MuiTypography-body2 css-1ncpc4g-MuiTypography-root"
  >
    Change the type of the second argument to "Address" to see an Escrow address
  </p>
</div>
`;
